packages/flutter_test/lib/src/binding.dart

```diff
@@ -700,13 +727,40 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     // The LiveTestWidgetsFlutterBinding overrides this to report the exception to the console.
   }
 
+  Future<dynamic> _handleMessage(dynamic mockMessage) async {
+    final Map<dynamic, dynamic> data =
+        (mockMessage as Map<dynamic, dynamic>)['data'] as Map<dynamic, dynamic>;
+    final String message = data['message'] as String;
+    final TextDirection textDirection = TextDirection.values[data['textDirection'] as int];
+    final dynamic assertivenessLevel = data['assertiveness'];
```

Comment:
You can avoid having to deal with `dynamic` and `null` by doing this:

```dart
final int assertivenessLevel = (data['assertiveness'] as int?) ?? 0;
```

This would also remove the need for the `if`/`else` below.